src/nnet3/nnet-simple-component.h

```diff
@@ -309,6 +310,7 @@ class AffineComponent: public UpdatableComponent {
   Component *CollapseWithNext(const FixedAffineComponent &next) const;
```

Comment:
These CollapseWithNext functions are not being used right now, as the calling code does not exist.  So leaving them out is fine-- it looks like they are not implemented anyway.  You can also remove any similar declarations in the PerElementScaleComponent that have not been implemented.